panda/src/navigation/navMeshBuilder.h

```diff
@@ -0,0 +1,181 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file navMeshBuilder.h
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+
+#ifndef NAVMESHBUILDER_H
+#define NAVMESHBUILDER_H
+
+#include "Recast.h"
+#include <string>
+#include "geom.h"
+#include "geomNode.h"
+#include "nodePath.h"
+#include "pandaFramework.h"
+#include "pandaSystem.h"
+#include "navMesh.h"
+
+enum SamplePolyAreas {
+  SAMPLE_POLYAREA_GROUND,
+  SAMPLE_POLYAREA_WATER,
+  SAMPLE_POLYAREA_ROAD,
+  SAMPLE_POLYAREA_DOOR,
+  SAMPLE_POLYAREA_GRASS,
+  SAMPLE_POLYAREA_JUMP,
+};
+enum SamplePolyFlags {
+  SAMPLE_POLYFLAGS_WALK = 0x01,		// Ability to walk (ground, grass, road)
+  SAMPLE_POLYFLAGS_SWIM = 0x02,		// Ability to swim (water).
+  SAMPLE_POLYFLAGS_DOOR = 0x04,		// Ability to move through doors.
+  SAMPLE_POLYFLAGS_JUMP = 0x08,		// Ability to jump.
+  SAMPLE_POLYFLAGS_DISABLED = 0x10,		// Disabled polygon
+  SAMPLE_POLYFLAGS_ALL = 0xffff	// All abilities.
+};
+struct BuildSettings {
+  
+  float cell_size;
+  float cell_height;
+  float agent_height;
+  float agent_radius;
+  float agent_max_climb;
+  float agent_max_slope;
+  float region_min_size;
+  float region_merge_size;
+  float edge_max_len;
+  float edge_max_error;
+  float verts_per_poly;
+  float detail_sample_dist;
+  float detail_sample_max_error;
+  int partition_type;
+  float nav_mesh_bMin[3];
+  float nav_mesh_bMax[3];
+  float tile_size;
+};
+
+
+class EXPCL_NAVIGATION NavMeshBuilder {
+PUBLISHED:
+  float get_actor_radius() { return _agent_radius; }
+  float get_actor_height() { return _agent_height; }
+  float get_actor_climb() { return _agent_max_climb; }
+  void set_actor_height(float h) { _agent_height = h; }
+  void set_actor_radius(float r) { _agent_radius = r; }
+  void set_actor_climb(float c) { _agent_max_climb = c; }
+  void set_partition_type(std::string p);
+  void reset_common_settings();
+  bool from_node_path(NodePath node);
```

Comment:
These names imply that you are creating a NavMeshBuilder from these inputs. Looking at the rest of this API, I believe `add_node_path()` and `add_geom()` may be more appropriate. Does it make sense to also expose `add_vertex()` and `add_triangle()`?

Reply:
The functions like set_actor_*() are optional. Those variables have a default value while from_node_path()/from_geom() is a compulsory function and is required to be called before build(). So, I thought to have different naming for them.

Reply:
Also, we may expose add_vertex() but that I don't think would make much sense as we cannot expose add_triangle(). Add_triangle() requires indices of 3 vertices as input. The user won't be knowing what index is assigned to what vertex.

Reply:
A convenience add_triangle method exposed to Python could surely take three LPoint3 instances, which are looked up in the vertex map (and if not present, added to the vertices).

Reply:
yes, will add that function as well

Reply:
I have added add_polygon method exposed to Python which takes in 3 LPoint3 instances.